src/components/DrawCore/index.tsx

```diff
@@ -254,1044 +255,1098 @@ const DrawCore = React.forwardRef<
 
-  const [drawRegion, setDrawRegion] = useState<Size | null>(null);
+    const [drawRegion, setDrawRegion] = useState<Size | null>(null);
 
-  const [originalImageSize, setOriginalImageSize] = useState<Size | null>(null);
+    const [originalImageSize, setOriginalImageSize] =
+      useState<Size | null>(null);
 
-  const [imageSize, setImageSize] = useState<Size | null>(null);
+    const [imageSize, setImageSize] = useState<Size | null>(null);
 
-  const drawContainer = useRef<View>(null);
+    const drawContainer = useRef<View>(null);
 
-  const viewShot = useRef<ViewShot>(null);
+    const viewShot = useRef<ViewShot>(null);
 
-  const [textVal, setTextVal] = useState<string>('');
+    const [textVal, setTextVal] = useState<string>('');
 
-  const currentItem = useSharedValue<DrawItem | null>(null);
+    const currentItem = useSharedValue<DrawItem | null>(null);
 
-  const initialItem = useSharedValue<DrawItem | null>(null);
+    const initialItem = useSharedValue<DrawItem | null>(null);
 
-  const [doneItems, setDoneItems] = useState<DrawItem[]>([]);
+    const [doneItems, setDoneItems] = useState<DrawItem[]>([]);
 
-  const textBaseHeight = useSharedValue<number | null>(null);
+    const [screenStates, setScreenStates] = useState<DrawItem[][]>([[]]);
 
-  const updateDoneItems = useCallback((done: DrawItem) => {
-    setDoneItems((previous) => previous.concat(done));
-  }, []);
+    const textBaseHeight = useSharedValue<number | null>(null);
 
-  useImperativeHandle(
-    ref,
-    () => ({
-      drawingContainer: drawContainer,
-      deleteSelectedItem: () => {
-        if (currentItem.value) {
-          currentItem.value = null;
-        }
-        onSelectionChange?.(false);
-      },
-      takeSnapshot: async (): Promise<string | undefined> => {
-        if (currentItem.value) {
-          updateDoneItems(currentItem.value);
-          currentItem.value = null;
-        }
-        return viewShot.current?.capture?.();
-      },
-    }),
-    [currentItem, onSelectionChange, updateDoneItems]
-  );
-
-  useEffect(() => {
-    mode.value = drawingMode;
-    if (currentItem.value) {
-      updateDoneItems(currentItem.value);
-    }
-    currentItem.value = null;
-    onSelectionChange?.(false);
-  }, [drawingMode, mode, currentItem, updateDoneItems, onSelectionChange]);
+    const updateDoneItems = useCallback((done: DrawItem) => {
+      setDoneItems((previous) => previous.concat(done));
+    }, []);
 
-  const addNewItem = runOnJS(updateDoneItems);
+    const updateScreenStates = useCallback((done: DrawItem[]) => {
+      const newScreenStates = screenStates;
+      newScreenStates.push(done);
+      setScreenStates(newScreenStates);
+    }, []);
 
-  const strokeWidth = useSharedValue<number>(2);
+    const deleteScreenStates = useCallback((indice: number) => {
+      const newScreenStates = screenStates;
+      newScreenStates.splice(indice, 1);
+      setScreenStates(newScreenStates);
```

Comment:
instead of using screenStates (and having to add it in hook dependencies), please use a method in setter

```
setScreenStates(previous => {
   previous.splice(indice, 1);
   return previous;
});
```